app/src/modules/settings/routes/data-model/field-detail/shared/related-collection-select.vue

```diff
@@ -18,13 +17,7 @@ const collectionExists = computed(() => {
 	return !!collectionsStore.getCollection(props.modelValue);
 });
 
-const availableCollections = computed(() => {
-	return orderBy(
-		collectionsStore.databaseCollections.filter((collection) => collection.meta),
-		['meta.sort', 'collection'],
-	);
-});
-
+const availableCollections = collectionsStore.configuredCollections;
```

Comment:
>It seems like non-table Groups are also present in the list when selecting a relational field, so somewhere we're not filtering those out as we should

It's here. I think this should be databaseCollections, no?

Reply:
Oh it looks like I messed up the definition for `configuredCollections`. It should be a filter on `databaseCollections` and not on `allCollections`.

Reply:
Well that fixes the non-table groups not filtering out out of the relational field selector, but then anything nested under a non-table group is filtered out of  the `/data-model` route 😅 

Before:
<img width="864" alt="Screenshot 2024-09-16 at 4 33 23 PM" src="https://github.com/user-attachments/assets/406a3590-cf6c-4f03-8871-3269e2f261c2">

After:
<img width="860" alt="Screenshot 2024-09-16 at 4 29 32 PM" src="https://github.com/user-attachments/assets/ef6beba7-a2e0-4d94-8d54-68f98e199c45">

Reply:
This is the culprit here:

https://github.com/directus/directus/blob/c216136deadc9f800cda0f37f4b613e3a91ae1a7/app/src/modules/settings/routes/data-model/collections/collections.vue#L34-L36

May want to update it to:

```
const rootCollections = computed(() => {
	return collectionsStore.allCollections.filter((collection) => !collection.meta?.group);
});
```

Or possibly define `rootCollections` in the collections store?